src/bidi-engine.js

```diff
@@ -0,0 +1,492 @@
+			_dir = _isInRtl ? DIR_RTL : DIR_LTR;
+			text = _invertString( text, sourceToTargetMap, levels );
+		} else if ( !_isInVisual && _isOutVisual && _isOutRtl ) {
+			/* LLTR->VRTL, LRTL->VRTL */
+			_dir = _isInRtl ? DIR_RTL : DIR_LTR;
+			text = _reorder( text, sourceToTargetMap, levels );
+			text = _invertString( text, sourceToTargetMap );
+		} else if ( _isInVisual && !_isInRtl && !_isOutVisual && !_isOutRtl ) {
+			/* VLTR->LLTR */
+			_dir = DIR_LTR;
+			text = _reorder( text, sourceToTargetMap, levels );
+		} else if ( _isInVisual && !_isOutVisual && ( _isInRtl ^ _isOutRtl ) ) {
+			/* VLTR->LRTL, VRTL->LLTR */
+			text = _invertString( text, sourceToTargetMap );
+			if ( _isInRtl ) { //LLTR -> VLTR
+				_dir = DIR_LTR;
+				text = _reorder( text, sourceToTargetMap, levels );
+			} else { //LRTL -> VRTL
+				_dir = DIR_RTL;
+				text = _reorder( text, sourceToTargetMap, levels );
+				text = _invertString( text, sourceToTargetMap );
+			}
+		} else if ( _isInVisual && _isInRtl && !_isOutVisual && _isOutRtl ) {
+			/*  VRTL->LRTL */
+			_dir = DIR_RTL;
+			text = _reorder( text, sourceToTargetMap, levels );
+			text = _invertString( text, sourceToTargetMap );
+		} else if ( !_isInVisual && !_isOutVisual && ( _isInRtl ^ _isOutRtl ) ) {
+			/* LRTL->LLTR, LLTR->LRTL */
+			var isSymmetricSwappingOrig = _isSymmetricSwapping;
+			if ( _isInRtl ) { //LRTL->LLTR
+				_dir = DIR_RTL;
+				text = _reorder( text, sourceToTargetMap, levels );
+				_dir = DIR_LTR;
+				_isSymmetricSwapping = false;
+				text = _reorder( text, sourceToTargetMap, levels );
+				_isSymmetricSwapping = isSymmetricSwappingOrig;
+			} else { //LLTR->LRTL
+				_dir = DIR_LTR;
+				text = _reorder( text, sourceToTargetMap, levels );
+				text = _invertString( text, sourceToTargetMap );
+				_dir = DIR_RTL;
+				_isSymmetricSwapping = false;
+				text = _reorder( text, sourceToTargetMap, levels );
+				_isSymmetricSwapping = isSymmetricSwappingOrig;
+				text = _invertString( text, sourceToTargetMap );
+			}
+		}
+		return text;
+	};
+
+/**
+ * setOptions( options )
+ *
+ * Sets options for Bidi conversion
+ *
+ * @options [Object]:
+ * - isInputVisual [boolean] (defaults to false):
+ * allowed values: true(Visual mode), false(Logical mode)
+ * - isInputRtl [boolean]:
+ * allowed values true(Right-to-left direction), false (Left-to-right directiion),
+ * undefined(Contextual direction, i.e.direction defined by first strong character of input string)
+ * - isOutputVisual [boolean] (defaults to false):
+ * allowed values: true(Visual mode), false(Logical mode)
+ * - isOutputRtl [boolean]:
+ * allowed values true(Right-to-left direction), false (Left-to-right directiion),
+ * undefined(Contextual direction, i.e.direction defined by first strong character of input string)
+ * - isSymmetricSwapping [boolean] (defaults to false):
+ * allowed values true(needs symmetric swapping), false (no need in symmetric swapping),
+ */
+	this.$bidiEngine.setOptions = function( options ) {
```

Comment:
I wanted to signify the exposed scope by this, if you don't like this, I can remove '$'